editortask.cpp

```diff
@@ -444,14 +444,18 @@ void EditorTask::menuRun()
 {
     if(filepath == "")
     {
-        dialog_task.showMessage("You have to save the file first!");
-        return;
-    }
-
-    const char* argv[] = {filepath.c_str()};
-    key_hold_down = true;
-
-    nl_exec(settings_task.mpython_path.content().c_str(), sizeof(argv), const_cast<char**>(argv));
+		FILE *f = fopen("/documents/tmp.py", "wb");
+		if(f)
```

Comment:
Does  it still make sense to run the file if opening or writing fails?

Reply:
Nope, I'll add a message to inform the user if the save has failed, it should be better and would be consistent with the menuSaveAs() routine imho

Reply:
There was some `saveAndRun` code before, that might be useful here

Reply:
Yup, found it there: https://github.com/Vogtinator/pyWrite/commit/a204ee54ca4067d5c6ce9c5babe619ff8e65f49f#diff-677b969513424dad8457dcbb75529942L174 my code should be better thanks to your advices now